src/layouts/components/Homepage/HeroBody.tsx

```diff
@@ -37,6 +42,276 @@ export interface HeroBodyFormFields {
+  } & HeroBodyFormFields
+}
+
+const HeroCenteredLayout = ({
+  title,
+  subtitle,
+  background,
+  index,
+  errors,
+}: HeroCenteredLayoutProps) => {
+  const { onChange } = useEditableContext()
+  return (
+    <>
+      <FormControl isRequired isInvalid={!!errors.title}>
+        <FormLabel>Hero title</FormLabel>
+        <Input
+          placeholder="Hero title"
+          id={`section-${index}-hero-title`}
+          value={title}
+          onChange={onChange}
+        />
+        <FormErrorMessage>{errors.title}</FormErrorMessage>
+      </FormControl>
+      <FormControl isRequired isInvalid={!!errors.subtitle}>
+        <FormLabel>Hero subtitle</FormLabel>
+        <Input
+          placeholder="Hero subtitle"
+          id={`section-${index}-hero-subtitle`}
+          value={subtitle}
+          onChange={onChange}
+        />
+        <FormErrorMessage>{errors.subtitle}</FormErrorMessage>
+      </FormControl>
+      <Box w="100%">
+        {/* TODO: migrate this to design system components */}
+        <FormContext
+          hasError={!!errors.background}
+          onFieldChange={onChange}
+          isRequired
+        >
+          <Box mb="0.5rem">
+            <FormTitle>Hero background image</FormTitle>
+          </Box>
+          <FormFieldMedia
+            value={background}
+            id={`section-${index}-hero-background`}
+            inlineButtonText="Select"
+          />
+          <FormError>{errors.background}</FormError>
+        </FormContext>
+      </Box>
+    </>
+  )
+}
+
+const HeroImageOnlyLayout = ({
+  errors,
+  background,
+  index,
+}: Omit<HeroCenteredLayoutProps, "subtitle" | "title">) => {
+  const { onChange } = useEditableContext()
+  return (
+    <Box w="100%">
+      {/* TODO: migrate this to design system components */}
+      <FormContext
+        hasError={!!errors.background}
+        onFieldChange={onChange}
+        isRequired
+      >
+        <Box mb="0.5rem">
+          <FormTitle>Hero background image</FormTitle>
+        </Box>
+        <FormFieldMedia
+          value={background}
+          id={`section-${index}-hero-background`}
+          inlineButtonText="Select"
+        />
+        <FormError>{errors.background}</FormError>
+      </FormContext>
+    </Box>
+  )
+}
+
+type SectionSize = "half" | "one-third"
+
+type SectionAlignment = "left" | "right"
+
+type SectionBackgroundColor = "black" | "white" | "translucent gray"
+
+interface HeroSideSectionProps extends HeroCenteredLayoutProps {
+  background: string
+  index: number
+  errors: {
+    title: string
+    subtitle: string
+    background: string
+  } & HeroBodyFormFields
+  size: SectionSize
+  alignment: SectionAlignment
+  backgroundColor: SectionBackgroundColor
+}
+
+const HeroSideSectionLayout = ({
+  background,
+  index,
+  errors,
+  title,
+  subtitle,
+  size = "half",
+  alignment = "left",
+  backgroundColor = "black",
+}: HeroSideSectionProps) => {
+  const [, setSectionSize] = useState(size)
```

Comment:
why do we need the set state without declaring the state?

Reply:
ps, i only added the panel - this means that the preview stuff here isn't working (it's a stop gap).

the actual preview component for these are gonna be added in a follow up and htis will be removed